extract/bbox.py

```diff
@@ -20,28 +20,27 @@ def __init__(self, upper_left_coordinate=None, lower_right_coordinate=None):
         self.lower_right_coordinate = lower_right_coordinate
 
 
-class BBoxMixin(object):
-    """Mixin class for node types which contain a BBox.
+class InvalidBboxString(Exception):
+    pass
 
-    The coordinates for the bbox should be contained in
-        self.attr['bbox']. The coordinates are assumed to be in the form
-        of a comma-seperated string like so: "x0, y0, x1, y1"
 
-    Adds a bbox read-only bbox property.
-    
-    (See https://docs.python.org/3/library/functions.html#property for more about
-    properties, and a description of how to make this property read/write if desired)."""
+def bbox_from_string(bbox_string):
+    points = bbox_string.split(",")
 
-    @property
-    def bbox(self):
-        if 'bbox' not in self.attr:
-            return None
+    if len(points) is not 4:
+        raise InvalidBboxString
+
+    upper_left_coordinate = Coordinate(float(points[0]), float(points[1]))
+    lower_right_coordinate = Coordinate(float(points[2]), float(points[3]))
 
-        points = self.attr['bbox'].split(",")
+    return BBox(upper_left_coordinate, lower_right_coordinate)
 
-        # All coordinates are cast to floats, which potentially introduces a loss of precision.
-        # TODO: investigate whether it would make more sense to use decimal.Decimal to preserve precision
 
-        upper_left_coordinate = Coordinate(float(points[0]), float(points[1]))
-        lower_right_coordinate = Coordinate(float(points[2]), float(points[3]))
-        return BBox(upper_left_coordinate, lower_right_coordinate)
\ No newline at end of file
+def bbox_from_node_attrs(attr):
```

Comment:
I think I'm going to move all these constructors (the type which takes XML elements) into their own thing